include/swift/AST/DiagnosticsSIL.def

```diff
@@ -371,6 +371,23 @@ ERROR(possible_return_before_yield, none,
 
 NOTE(one_yield, none, "yield along one path is here", ())
 
+NOTE(false_branch_doesnt_yield, none,
+    "must yield when the condition is false", ())
+
+NOTE(true_branch_doesnt_yield, none,
```

Comment:
Instead of duplicating the entire diagnostic text for the true/false case you can use "when the condition is %select{false|true}0"